vllm/spec_decode/metrics.py

```diff
@@ -77,13 +78,20 @@ def __init__(self,
         self._rejsample_metrics_collect_interval_s = collect_interval_s
         self._last_metrics_collect_time = self._timer()
 
-    def init_gpu_tensors(self, rank: int) -> None:
+    def init_tensors(self, rank: int, device: str) -> None:
         self._rank = rank
-        self._copy_stream = torch.cuda.Stream()
+        if 'hpu' == device.type:
+            import habana_frameworks.torch as htorch
+            self._copy_stream = htorch.hpu.Stream()
+        else:
+            self._copy_stream = torch.cuda.Stream()
 
     def maybe_collect_rejsample_metrics(
             self, k: int) -> Optional[SpecDecodeWorkerMetrics]:
 
+        if not current_platform.is_cuda_alike():
+            return None
```

Comment:
@michalkuligowski , for non_cuda backend, this function is not usable because it hard code with `torch.cuda.Event`
=> This will not affect workflow, so I think we might skip it to make functionality ready?

I provide an example as below
```
    def _collect_rejsample_metrics(
            self, k: int,
            ready_event: torch.cuda.Event) -> SpecDecodeWorkerMetrics:
```

```
def _copy_rejsample_metrics_async(self) -> torch.cuda.Event:
        """Copy rejection/typical-acceptance sampling metrics
        (number of accepted tokens, etc) to CPU asynchronously.

        Returns a CUDA event recording when the copy is complete.
        """
        assert self._copy_stream is not None
        self._copy_stream.wait_stream(torch.cuda.current_stream())

        with torch.cuda.stream(self._copy_stream):
            self._aggregate_num_accepted_tokens.copy_(
                self.spec_decode_sampler.num_accepted_tokens,
                non_blocking=True)
            self._aggregate_num_emitted_tokens.copy_(
                self.spec_decode_sampler.num_emitted_tokens, non_blocking=True)
            # Number of draft tokens is calculated on CPU, so no copy is
            # required.
            self._aggregate_num_draft_tokens = (
                self.spec_decode_sampler.num_draft_tokens)

        aggregate_metrics_ready = torch.cuda.Event()
        aggregate_metrics_ready.record(self._copy_stream)

        return aggregate_metrics_ready
```